packages/native-stack/src/types.tsx

```diff
@@ -378,6 +378,15 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   gestureEnabled?: boolean;
+  /**
+   * Whether the dismissal of the screen on tap of the header back button on Android should be performed by the native or the
+   * navigation's side. Defaults to `false`.
+   *
+   * Only supported on Android.
+   *
+   * @platform android
+   */
+  nativeBackButtonDismissalEnabled?: boolean;
```

Comment:
Is there any use case for exposing this to the user in React Navigation? Why not have it always set to `false`?

Reply:
When you use the native back button (`nativeBackButtonDismissalEnabled: true`) native changes can go through to the JS side because a view on the JS side is detached after the transition. It is necessary for eg. animations and transition context.

https://user-images.githubusercontent.com/39658211/154917853-ba437a1c-9176-44b4-be76-4d67a9a15152.mov


When using navigation's go back (`nativeBackButtonDismissalEnabled: false`) these changes can't be accessed because the view is detached immediately on the JS side.

https://user-images.githubusercontent.com/39658211/154917877-99a4fefa-953b-4d0e-bbfd-b3c29efd2d86.mov